docs(chatbot): add component pages (#244)
siemens/ix-docs: docs/components/chat-input/guide.md

- **Follow-up slot:** Optionally include a slot for follow-up questions. We typically use tertiary outline [buttons](../button) or [icon buttons](../icon-button) for that purpose.
- **Attachment slot:** Optionally include a slot for [attachments](../chat-attachment/).
- **Start slot:** Add secondary actions. We recommend using tertiary outline [icon buttons](../icon-button) and trying to stick to one action. If you have more than one action, use [dropdown buttons](../dropdown-button/).
- **End slot:** Use this slot to add alternative input methods, e.g. voice input.
- **Disclaimer:** In AI contexts, we recommend including a visible disclaimer under the input instead of under each [AI message](../ai-message/). For Siemens AG products, find legal disclaimers [here](https://code.siemens.com/siemens-ix/ix-brand-theme/-/blob/main/apps/documentation/src/pages/legal-disclaimers-copilots.md).
- **Character limit:** Set soft and hard character limits that either warn users or prevent further input.

## Behavior in context

- **Interaction:** Keep the input compact at start and let it grow to multiline while users type
- **Overflow:**

>     - On the follow-up slot: If the follow-up actions exceed max width, they break into multiple lines

Reviewer: "On the follow-up" is sufficient, "slot" is unnecessarily technical here and also not used in bullet point 3 where it says "On the attachments"

    - On the input: If the input reaches max height, the input shows a vertical scroll
    - On the attachments: If the attachments reach max width, they show a horizontal scroll

![Overflow behavior](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7940-7020&t=mrtbkWj76QJhvNLI-11)

## States

> Chat inputs have six states: default, hover, focused and processing.

Reviewer: align: six states are announced, four states are named, three states are shown in the illustration

Reviewer 2: So true. The listed states here were correct - update the illustration and the number shown here

---
doc-type: 'tab-item'

> description: 'Usage guide for chat input'

Reviewer 3: full stop?

---

> # Chat input - Usage

Reviewer 3: Question: this was named "prompt input field" by brand. how can we deal with this conflict?

Reviewer 2: We followed Element's approach to have the component more generic and usable for all chat contexts, not specifically only for AI chatbots - that's why we also aligned the naming with them here.

That's also why we think it's not a conflict, but our DS offers generic components (chat input) that can then be used for specific use cases (as a prompt input)

cc: @flxlst09

Reviewer: Yes, should not be limited to prompts :)

> Chat inputs help users compose and send short conversational messages in chat experiences. We recommend using them for quick, iterative exchanges, not for multi-step data entries.

Reviewer 3: ```suggestion
Users write and send their prompts (queries, commands, tasks, etc.) in chat inputs. 
```
Note people can use them for whatever they want. Some copilots allow them to add PLC engineering steps in one field. Can we also link to the placeholder text here? https://ix.siemens.io/docs/guidelines/conversational-design/essentials/wording-terms

Reviewer 2: As mentioned, we try to have this component open for any chat-based interaction. Could we see an example of these PLC steps? 

About the link: I'd add the link to the option section, where we describe the placeholder.

> Chat inputs help users compose and send short conversational messages in chat and assistant experiences. We recommend using them for quick, iterative exchanges, not for long-form forms or multi-step data entry.

Reviewer: long-form forms is repetitive

![Anatomy component](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7940-7427&t=mrtbkWj76QJhvNLI-11)

> 1. Follow-up questions

Reviewer 3: question: I think this is prompt card / suggested prompts not followup questions as they can appear before you even ask anything.

Reviewer 2: Not sure if I understood correctly - here's the internal guideline: https://www.figma.com/design/KbgPxj7qLgngXkJfnDM4Ty/SDL-AI-UX-Guidelines?node-id=225-51247&t=pbv0c2rQagXjKN4E-11

Reviewer 2: @flxlst09 @danielleroux discuss whether calling them follow-up _question_ is right since it's not only questions - it can be also prompt card / suggested prompts.

2. Attachments
3. Textarea
4. Start slot for secondary actions
5. Disclaimer
6. End slot for alternative input methods
7. Send button

## Options

> - **Placeholder:** Use clear, contextual prompts, e.g. “Enter a command, question or topic…”

Reviewer 3: https://ix.siemens.io/docs/guidelines/conversational-design/essentials/wording-terms 
```suggestion
- **Placeholder:** For Siemens AI products, we have a set of brand-aligned generic, non-specific placeholders to choose from to encourage focused prompts.   
```https://ix.siemens.io/docs/guidelines/conversational-design/essentials/wording-terms

Reviewer 2: I'm confused ... are these placeholder texts now for Siemens only or public?

If they are not and Siemens only -> we should move them to the inner source project
If they are public -> what's your suggested change about?

> - **Follow-up slot:** Optionally include a slot for follow-up questions. We typically use tertiary outline [buttons](../button) or [icon buttons](../icon-button) for that purpose.

Reviewer 3: question: i dont understand what this is

Reviewer 2: Maybe the internal guideline helps: https://www.figma.com/design/KbgPxj7qLgngXkJfnDM4Ty/SDL-AI-UX-Guidelines?node-id=225-51247&t=pbv0c2rQagXjKN4E-11

> - **Attachment slot:** Optionally include a slot for [attachments](#).

Reviewer: Link missing

- **Start slot:** Add secondary actions. We recommend using tertiary outline [icon buttons](../icon-button) and trying to stick to one action. If you have more than one action, use [dropdown buttons](../dropdown-button/).
- **End slot:** Use this slot to add alternative input methods, e.g. voice input.

> - **Disclaimer:** In AI contexts, include a short visible disclaimer near the input.

Reviewer 3: add link for internal disclaimers: https://code.siemens.com/siemens-ix/ix-brand-theme/-/blob/main/apps/documentation/src/pages/legal-disclaimers-copilots.md

> - **Disclaimer:** In AI contexts, include a short visible disclaimer near the input.

Reviewer 3: ```suggestion
- **Disclaimer:** In AI contexts, we recommend including a visible disclaimer under the chat input instead of under each AI message. For Siemens products, the legal disclaimers are found here. 
```

- **Character limit:** Set soft and hard character limits that either warn users or prevent further input.

## Behavior in context

- **Interaction:** Keep the input compact at start and let it grow to multiline while users type
- **Overflow:**
  - On the follow-up slot: If the follow-up actions exceed max width, they break into multiple lines
  - On the input: If the input reaches max height, the input shows a vertical scroll
  - On the attachments: If the attachments reach max width, they show a horizontal scroll

![Anatomy component](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7940-7020&t=mrtbkWj76QJhvNLI-11)

## States

Chat inputs have six states: default, hover, focused and processing.

> ![States](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7940-2809&t=mrtbkWj76QJhvNLI-4)

Reviewer: the illustration for entering text should show the focus outline, shouldn't it?

Reviewer 2: good catch 😂

## Dos and Don’ts

> - Do keep labels and helper text explicit instead of relying on placeholder text only

Reviewer 3: link to https://www.figma.com/design/lqjt9c5IzzwQ4eJ4nqG7Kv/AI-Terminology?node-id=1-9&t=d5UkOPKJfj9qDmYM-1

Reviewer 2: Actually, now thinking about it, this do makes 0 sense since the chat input has neither label nor helper text 😂 I'll remove it altogether.

> - Do keep action labels verb-based and specific so users understand the outcome

Reviewer 3: same https://www.figma.com/design/lqjt9c5IzzwQ4eJ4nqG7Kv/AI-Terminology?node-id=1-9&t=d5UkOPKJfj9qDmYM-1

Reviewer 2: do we have the terms in our gitlab repo as well? - people without Figma access won't be able to access this document

> - Do show attachment or action context before send so users can review what they submit

Reviewer 3: hard to follow and understand

Reviewer 2: true, I'll just remove it since it's anyways common sense (I hope 😂 )

> - Do keep AI disclaimers concise and tied to AI generation contexts

Reviewer 3: ```suggestion
```

> - Don’t allow sending empty input

Reviewer 3: ```suggestion
- Don’t allow users to send empty prompt input fields. 
```

Reviewer 2: I think it's either submit [...] input field **or** send [...] input? 🤔 - I'll change it to the second option but leave it open

- Don’t rely on color alone to communicate validation errors

> - Don’t hide essential guidance only in tooltips

Reviewer 3: not sure this is even relevant for this - would delete

> - Don’t use AI disclaimer content in non-AI messaging contexts

Reviewer 3: ```suggestion
- Don’t place AI disclaimers away from user input and messages
```
